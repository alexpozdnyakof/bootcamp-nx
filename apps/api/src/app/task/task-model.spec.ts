import TaskModel from './task-model'
describe('TaskModel', () => {
	it('should return all tasks', async () => {
		await expect(TaskModel.GetAll()).resolves.toMatchSnapshot()
	})

	it('should return task with id 1', async () => {
		await expect(TaskModel.GetOne(1)).resolves.toMatchSnapshot()
	})

	it('should throw error when get non-existing task', async () => {
		await expect(TaskModel.GetOne(10)).rejects.toEqual(
			new Error('Not Found')
		)
	})
	it('should create new one task', async () => {
		const dto = {
			title: '新しい計画',
			done: false,
		}
		try {
			const pre = await TaskModel.GetAll()
			await TaskModel.Add(dto)
			const past = await TaskModel.GetAll()

			expect(past).toHaveLength(pre.length + 1)
			expect(past[past.length - 1]).toMatchSnapshot({
				created: expect.any(String),
				updated: expect.any(String),
				done: false,
				title: '新しい計画',
				id: 4,
			})
		} catch (e) {
			console.log(e)
		}
	})
	it('should delete task with id 3', async () => {
		const pre = await TaskModel.GetAll()
		await TaskModel.Delete(3)

		const past = await TaskModel.GetAll()
		expect(past).toHaveLength(pre.length - 1)
		await expect(TaskModel.GetOne(3)).rejects.toEqual(
			new Error('Not Found')
		)
	})

	it('should rejects when delete non-existing task ', async () => {
		await expect(TaskModel.Delete(10)).rejects.toEqual(
			new Error('Not Found')
		)
	})

	it('should update entity with id 1', async () => {
		const dto = {
			title: '新しい計画',
			done: true,
		}

		let { title, done } = await TaskModel.GetOne(1)

		expect({ title, done }).not.toEqual(dto)

		await TaskModel.Update(1, dto)
		;({ title, done } = await TaskModel.GetOne(1))

		expect({ title, done }).not.toEqual(dto)
	})

	it('should rejects when update non-existing task ', async () => {
		const dto = {
			title: '新しい計画',
			done: false,
		}

		await expect(TaskModel.Update(10, dto)).rejects.toEqual(
			new Error('Not Found')
		)
	})
})
