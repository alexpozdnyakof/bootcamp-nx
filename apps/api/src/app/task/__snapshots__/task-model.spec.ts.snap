// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskModel should create new one task 1`] = `
Object {
  "created": Any<String>,
  "done": false,
  "id": 4,
  "title": "新しい計画",
  "updated": Any<String>,
}
`;

exports[`TaskModel should return all tasks 1`] = `
Array [
  Object {
    "created": "2022-11-21T11:21:14.543Z",
    "done": false,
    "id": 1,
    "title": "血液レポートのグラフが空白になっている",
    "updated": "2022-11-21T11:21:14.543Z",
  },
  Object {
    "created": "2022-11-21T11:21:14.544Z",
    "done": true,
    "id": 2,
    "title": "無効にする|| ユーザーがアカウントを無効にできない",
    "updated": "2022-11-21T11:21:14.544Z",
  },
  Object {
    "created": "2022-11-21T11:21:14.545Z",
    "done": false,
    "id": 3,
    "title": "プロフィール、プロフィールの編集、ポップアップ",
    "updated": "2022-11-21T11:21:14.545Z",
  },
]
`;

exports[`TaskModel should return task with id 1 1`] = `
Object {
  "created": "2022-11-21T11:21:14.543Z",
  "done": false,
  "id": 1,
  "title": "血液レポートのグラフが空白になっている",
  "updated": "2022-11-21T11:21:14.543Z",
}
`;
